source/MaterialXGenShader/Util.cpp

```diff
@@ -13,119 +13,156 @@ namespace MaterialX
+            {
+                return true;
+            }
         }
         return false;
     }
 
-    bool isOne(float v)
-    {
-        return v > EPS_ONE;
-    }
+    using OpaqueTestPair = std::pair<string, float>;
+    using OpaqueTestPairList = vector<OpaqueTestPair>;
 
-    bool isOne(const Color3& c)
+    // Get corresponding input for an interfacename for a nodegraph. 
+    // The check is done for any corresponding nodedef first and then for 
+    // any direct child input of the nodegraph.
+    InputPtr getInputInterface(const string& interfaceName , NodePtr node)
     {
-        return isOne(c[0]) && isOne(c[1]) && isOne(c[2]);
+        InputPtr interfaceInput = nullptr;
+        ElementPtr parent = node->getParent();
+        NodeGraphPtr nodeGraph = parent ? parent->asA<NodeGraph>() : nullptr;
+        if (nodeGraph)
+        {
+            NodeDefPtr nodeDef = nodeGraph->getNodeDef();
+            if (nodeDef)
+            {
+                interfaceInput = nodeDef->getInput(interfaceName);
+            }
+            else
+            {
+                interfaceInput = nodeGraph->getInput(interfaceName);
+            }
+        }
+        return interfaceInput;
     }
 
-    bool isOne(ValuePtr value)
+    bool hasTransparentInputs(const OpaqueTestPairList& opaqueInputList, NodePtr node)
     {
-        if (value->isA<float>() && isOne(value->asA<float>()))
-        {
-            return true;
-        }
-        if (value->isA<Color3>() && isOne(value->asA<Color3>()))
+        for (auto opaqueInput : opaqueInputList)
         {
-            return true;
+            InputPtr interfaceInput = node->getInput(opaqueInput.first);
+            if (interfaceInput)
+            {
+                if (interfaceInput->getConnectedNode())
+                {
+                    return true;
+                }
+                ValuePtr value = interfaceInput->getValue();
+                if (value && !isEqual(value, opaqueInput.second))
+                {
+                    return true;
+                }
+            }
         }
         return false;
     }
 
-    bool isTransparentShaderNode(NodePtr node)
+    bool isTransparentShaderNode(NodePtr node, NodePtr interfaceNode)
     {
         if (!node || node->getType() != SURFACE_SHADER_TYPE_STRING)
         {
             return false;
         }
 
-        // Check opacity
-        InputPtr opacity = node->getActiveInput("opacity");
-        if (opacity)
+        // Inputs on a surface shader which are checked for transparency
+        const OpaqueTestPairList inputPairList = { {"opacity", ONE_VALUE},
+                                                   {"existence", ONE_VALUE},
+                                                   {"transmission", ZERO_VALUE} };
+
+
+        // Check against the interface if a node is passed in to check against
+        OpaqueTestPairList interfaceNames;
+        if (interfaceNode)
```

Comment:
Check the instance of the functional graph's node first before the graph itself as it's value/connection overrides any default ones.